aptos-move/aptos-vm/src/aptos_vm.rs

```diff
@@ -877,6 +880,109 @@ impl AptosVM {
             traversal_context,
         )
     }
+    fn execute_automation_transaction<'a, 'r, 'l>(
+        &'l self,
+        resolver: &'r impl AptosMoveResolver,
+        mut session: UserSession<'r, 'l>,
+        gas_meter: &mut impl AptosGasMeter,
+        traversal_context: &mut TraversalContext<'a>,
+        txn_data: &TransactionMetadata,
+        payload: &'a AutomationTransactionPayload,
+        log_context: &AdapterLogSchema,
+        new_published_modules_loaded: &mut bool,
+        change_set_configs: &ChangeSetConfigs,
+    ) -> Result<(VMStatus, VMOutput), VMStatus> {
+        fail_point!("aptos_vm::execute_automation_transaction", |_| {
+            Err(VMStatus::Error {
+                status_code: StatusCode::UNKNOWN_INVARIANT_VIOLATION_ERROR,
+                sub_status: Some(move_core_types::vm_status::sub_status::unknown_invariant_violation::EPARANOID_FAILURE),
+                message: None,
+            })
+        });
+        if !payload.is_valid() {
+            return Err(VMStatus::error(
+                StatusCode::INVALID_AUTOMATION_PAYLOAD,
+                Some(AutomationTransactionPayload::entry_function_reference()),
+            ));
+        }
+
+        gas_meter.charge_intrinsic_gas_for_transaction(txn_data.transaction_size())?;
+        if txn_data.is_keyless() {
+            gas_meter.charge_keyless()?;
+        }
+
+        match payload {
+            AutomationTransactionPayload::EntryFunction(entry_fn) => {
+                session.execute(|session| {
+                    self.validate_automation_txn_inner_payload(
+                        session,
+                        gas_meter,
+                        txn_data.senders(),
+                        entry_fn,
+                    )
+                })?;
+                session.execute(|session| {
+                    self.validate_and_execute_entry_function(
+                        resolver,
+                        session,
+                        gas_meter,
+                        traversal_context,
+                        txn_data.senders(),
+                        entry_fn,
+                        txn_data,
+                    )
+                })?
+            },
+            AutomationTransactionPayload::EntryFunctionArguments(args) => {
+                session.execute(|session| {
+                    self.validate_automation_txn_inner_payload_v2(
+                        session,
+                        gas_meter,
+                        txn_data.senders(),
+                        args.inner_payload(),
+                    )
+                })?;
```

Comment:
Ok, my understanding is that we are either going to keep this case, or the previous one (where we have to make an assumption that payload is always the first argument in both Rust and Move layer). I agree that such an assumption would make the design fragile, so it's best to contruct it from the arguments.